service/payment.go

```diff
@@ -111,31 +111,40 @@ func (service *PaymentService) CreatePaymentSession(w http.ResponseWriter, req *
 		}
 	}
 
-	var paymentResource models.PaymentResource
-	paymentResource.Data.CreatedBy = models.CreatedBy{
+	//  Create payment session REST data from writable input fields and decorating with read only fields
+	var paymentResourceRest models.PaymentResourceRest
+	paymentResourceRest.CreatedBy = models.CreatedByRest{
 		ID:       req.Header.Get("Eric-Identity"),
 		Email:    email,
 		Forename: forename,
 		Surname:  surname,
 	}
-	paymentResource.Data.Amount = totalAmount
+	paymentResourceRest.Amount = totalAmount
 	// To match the format time is saved to mongo, e.g. "2018-11-22T08:39:16.782Z", truncate the time
-	paymentResource.Data.CreatedAt = time.Now().Truncate(time.Millisecond)
+	paymentResourceRest.Costs = *costs
```

Comment:
maybe move this costs assignment down a line after the created at assignment due to the comment above relating to truncating that time

Reply:
Done